qiskit/circuit/quantumcircuit.py

```diff
@@ -1221,6 +1221,17 @@ def qasm(self, formatted=False, filename=None):
         else:
             return string_temp
 
+    def _ipython_display_(self):
+        if self.width() > 70 or self.size() > 100 or self.depth() > 30:
+            print(self.__repr__())
```

Comment:
It might not be worthy. Our drawers take the full circuit they do not support "partial" inputs. It could be possible to cut out the result after generation, but that will require several seconds for generation, which is the thing avoid.

At least my use case is for simple cases, like in demos.